libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt

```diff
@@ -604,7 +605,7 @@ class RustMatrixRoom(
     override fun getWidgetDriver(widgetSettings: MatrixWidgetSettings): Result<MatrixWidgetDriver> = runCatching {
         RustWidgetDriver(
             widgetSettings = widgetSettings,
-            room = innerRoom,
+            room = innerRoom as Room,
```

Comment:
This would need to be avoided too.

Reply:
I don't think this is possible either, same reason: Rust code expects to receive Rust concrete types, unless we exported this as a trait instead, maybe.